### PR TITLE
Actor resource backlog hotfix

### LINKED ISSUE #12441: [autoscaler] Actor resource demands are not cleared after actor is scheduled
### What is the problem?
* ray nightly

### Reproduction (REQUIRED)
Take the following steps:
1. start a k8s autoscaling cluster
2. Execute: (I did in in ipython in ray head container)
```
import os
import ray

ray.init(address="auto")

@ray.remote(num_cpus=0.2)
class ActorA:
    def __init__(self):
        pass

a = ActorA.remote()
b = ActorA.remote()

# wait untill deployed, then kill actors or exit ipython
ray.kill(a)
ray.kill(b)
```
-> output of autoscaling monitor:
```
2020-11-26 10:23:43,710 INFO resource_demand_scheduler.py:193 -- Resource demands: [{'CPU': 0.2}, {'CPU': 0.2}]
```
-> the actor resources stick, instead of getting cleaned up

I would expect the resources to be cleaned up...

- [x] I have verified my script runs in a clean environment and reproduces the issue.
- [x] I have verified the issue also occurs with the [latest wheels](https://docs.ray.io/en/master/installation.html).

### PR DESCRIPTION
**Pull request by wuisawesome**
## Why are these changes needed?

This fixes a bug in which GCS was (incorrectly) reporting a backlog size of -1 for actor leases, which broke the raylet's bookkeeping. 

## Related issue number

Closes #12441

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by wuisawesome** [comment]
cc @AmeerHajAli

**Comment by AmeerHajAli** [comment]
@wuisawesome can we add the test from #12441 ?

**Comment by wuisawesome** [comment]
This is Eric's test case.

**Comment by AmeerHajAli** [comment]
But what is wrong with adding the other test case too? Can we just copy paste it?

**Comment by wuisawesome** [comment]
I'm not sure what you mean. How would you suggest running the original reproduction in a test case?

**Comment by AmeerHajAli** [comment]
Can you try to see what happens if you kill the actor? Or if it was killed before being scheduled ?

**Comment by wuisawesome** [comment]
I mean it can be done, but I don't see how it's relevant to this PR/bug which is purely about a parameter not being set during actor creation.

**Comment by AmeerHajAli** [comment]
Can we please manually test it and make sure it does not happen? I know it might not be relevant, I just want to make sure this bug is resolved, and a reasonable way to know that is to rerun the faulty code and see if it is not failing.

File: src/ray/gcs/gcs_server/gcs_actor_scheduler.cc [hunk 1]
```diff
@@ -208,6 +209,9 @@ void GcsActorScheduler::LeaseWorkerFromNode(std::shared_ptr<GcsActor> actor,
   remote_address.set_ip_address(node->node_manager_address());
   remote_address.set_port(node->node_manager_port());
   auto lease_client = GetOrConnectLeaseClient(remote_address);
+  // Actor leases should be sent to the raylet immediately, so we should never build up a
+  // backlog in GCS.
+  int backlog_size = report_worker_backlog_ ? 0 : -1;
```
Comment (ericl) [inline src/ray/gcs/gcs_server/gcs_actor_scheduler.cc:214]: Why don't we always pass in 0?

**Review by ericl — APPROVED** [review]
The change seems to fix the problem, but do you know where the backlog size was being injected? I didn't know we had any code that did that in the GCS.

**Comment by ericl** [comment]
@AmeerHajAli I think the test case here supercedes the other one, since it is checking the resource is removed after scheduling. Killing the actor after that has no effect.

**Comment by ericl** [comment]
Let's merge and try this out in nightly wheels.

**Comment by wuisawesome** [comment]
> where the backlog size was being injected?

Default argument here: where the backlog size was being injected?

**Comment by ericl** [comment]
Can you provide a link to the code site? I can't find where the GCS gets a backlog.

**Comment by wuisawesome** [comment]
Oh whoops, sorry  😅

https://github.com/ray-project/ray/blob/master/src/ray/raylet_client/raylet_client.h#L68

**Comment by ericl** [comment]
Hmm I see it defaults to -1, but why does that cause the backlog size to build up forever if not set to zero? Do we understand where that accumulated value is coming from?

**Comment by wuisawesome** [comment]
Yeah, in AppendTask, we check if the backlog size is non-negative before appending. https://github.com/ray-project/ray/blob/master/src/ray/raylet/scheduling_queue.cc#L76

In RemoveTask, we blindly subtract. https://github.com/ray-project/ray/blob/master/src/ray/raylet/scheduling_queue.cc#L98

**Comment by ericl** [comment]
Thanks, good find there!

**Pull request merged**